packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -1150,22 +1150,43 @@ export class Vector3 {
      * @param result defines the Vector3 where to store the result
      */
     public projectOnPlaneToRef(plane: Plane, origin: Vector3, result: Vector3): void {
-        const n = plane.normal;
-        const d = plane.d;
-
-        const V = MathTmp.Vector3[0];
-
-        // ray direction
-        this.subtractToRef(origin, V);
+        // Use the normal scaled to the plane offset as the origin for the formula
+        const planeOrigin = MathTmp.Vector3[0];
+        plane.normal.scaleToRef(-plane.d, planeOrigin)
+
+        // Since the normal in Babylon should point toward the viewer, invert it for the dot product
+        const inverseNormal = MathTmp.Vector3[1];
+        plane.normal.negateToRef(inverseNormal);
+        
+        // This vector is the direction
+        const direction = this;

```

Comment:
why is that needed?

Reply:
There's no reason, it's just a habit of mine to name things in a way that makes it as clear as possible what the code is doing. I believe it should be optimized out so it's purely aesthetic.

Reply:
Let me know if I should change this or if it's good as is. :+1: